Issue/oak 8848

shodaaan opened the pull request.

adelina-marian committed:
OAK-8848 added unit test that triggers jcr:versionHistory exception

shodaaan committed:
- modified VersionEditor to allow replacing a deleted node with a mix:versionable node within the same session

shodaaan commented:
- included the unit test created by Adelina Marian, which reproduces the bug
- added a fix to VersionEditor that allows a node with mix:versionable to be moved over a deleted node

shodaaan committed:
- modified VersionEditor to allow replacing a deleted node with a mix:versionable node within the same session

shodaaan committed:
- marked new unit test as @Ignore in order to check Oak Jenkins build

shodaaan committed:
- marked new unit test as @Ignore in order to check Oak Jenkins build

shodaaan committed:
Merge branch 'apache:trunk' into issue/OAK-8848

Commit:
OAK-10813 : using projection while listing/removing root temp nodes

Commit:
OAK-10813 : using projection while listing children of an node

stefan-egli reviewed (commented):
Would it be possible to try to reproduce a constraint violation exception when replacing a node with another node when both have a version history? (to see if this is can be reproduced - and is the same or a different issue)

Commit:
OAK-10823 : replaced db.collection.remove with db.collection.deleteOne/deleteMany to make it compatible with both mongo & mongosh

reschke committed:
OAK-10824: Release Oak 1.64.0 - Candidate Release Notes

ahanikel committed:
OAK-10771 - Add disk cache size stats and issue warning if evicted segment has zero length (#1427)

* OAK-10771 - Log if evicted segment has length zero.

* OAK-10771 - Add diskCacheSize stats

* OAK-10771 - Record cache size and change

---------

Co-authored-by: Axel Hanikel <ahanikel@adobe.com>

Commit:
OAK-10826 - Misleading warning about missing checkpoint

Commit:
OAK-10823 : reverted un-intended changes from var to let/const

reschke committed:
OAK-10824: Release Oak 1.64.0 - Candidate Release Notes

reschke committed:
[maven-release-plugin] prepare release jackrabbit-oak-1.64.0

reschke committed:
[maven-release-plugin] prepare for next development iteration

nfsantos committed:
OAK-10789 - Log paths used for inclusing/exclusion for Mongo regex filters in job summary (#1478)

fabriziofortino committed:
OAK-10811: (oak-search-elastic) reduce contention in IndexTracker (#1466)

* OAK-10811: (oak-search-elastic) reduce contention in IndexTracker

* OAK-10811: ElasticIndexInfoProvider uses node api to get :status info

* OAK-10811: rolled back unneeded change

t-rana committed:
Issues/OAK-10675 (#1385)

* OAK-10675: add service principal support in oak-blob-cloud-azure

* OAK-10675: use user delegation key signed sas for service principal and add license to new files

* OAK-10675: some self review changes

* OAK-10675: set null blob headers as empty string

* OAK-10675: update pom to resolve package dependency

* OAK-10675: add import packages to resolve bundle

* OAK-10770: embed runtime dependencies of azure identity in oak-segment-azure

* OAK-10770: remove unused imports, refactor dependencies

* OAK-10770: add imports, refactor dependencies

* OAK-10675: fix runtime dependency issues, code review

reschke committed:
OAK-10829: examples: update build-helper-maven-plugin to 3.6.0 (#1483)

mbaedke commented:
> Would it be possible to try to reproduce a constraint violation exception when replacing a node with another node when both have a version history? (to see if this is can be reproduced - and is the same or a different issue)

@stefan-egli I don't understand, isn't that exactly what the new test does?

shodaaan committed:
- changed fix for moving versionable node over a deleted versionable node by checking that the versionable node was moved
- created unit tests classes for the OAK-8848 use cases using multiple fixtures: SEGMENT_TAR - base test, DOCUMENT_NS (Mongo), MEMORY_NS
- moved original unit tests from VersionManagementTest to the new test classes and added test for more complex node hierarchies

mbaedke committed:
OAK-10624: o.a.j.o.namepath.JcrPathParser does not accept some valid local names containing '{' or '}'. (#1285)

Fixed issue and added test case.

Co-authored-by: Julian Reschke <reschke@apache.org>

shodaaan committed:
Merge remote-tracking branch 'Oak_Jackrabbit_Horia/issue/OAK-8848' into OAK-8848

mbaedke committed:
OAK-6756: Convert oak-auth-external to OSGi R7 annotations (#1371)

Done

---------

Co-authored-by: Julian Sedding <jsedding@apache.org>

shodaaan commented:
@stefan-egli: I created another unit test where the 2 nodes have version histories and the code flow is the same and throws ConstraintViolationException.

So if the 2 nodes have only "mix:versionable" or version histories, the behaviour and result is the same. Apparently the "jcr:versionHistory" property is created automatically for mix:versionable nodes even if no check-out / check-in operations are performed on them, and the VersionEditor.propertyChanged method throws 
the exception for the after node's "jcr:versionHistory" property.

The fix I propose in the latest commit works for both cases and was tested in unit tests using Segment, DocumentNodeStore (Mongo) and MemoryNodeStore as underlying storage when running the unit tests.

t-rana committed:
OAK-10780: add azure access token refresh logic (#1441)

* OAK-10780: add azure access token refresh logic

* OAK-10780: modify log statement

* OAK-10780: modify log statement

* OAK-10780: reduce token refresh interval

reschke committed:
OAK-10830: Upgrade jackson-databind dependency to 2.17.1 (#1486)

reschke commented on oak-jcr/src/test/java/org/apache/jackrabbit/oak/jcr/MoveVersionableNodeWithMemoryRepositoryTest.java:
```diff
@@ -0,0 +1,26 @@
+package org.apache.jackrabbit.oak.jcr;
```
So, IIUC, you have created multiple test classes so that the different mixtures are covered automatically. That was certainly helpful for testing, but it wouldn't scale if we did this everywhere.

So I would propose to simplify that again; and we also should wortk on the orthogonal issue of documenting and potentially fixing the CI pipelines.

shodaaan replied:
fixed in latest commit

reschke committed:
Revert "OAK-10830: Upgrade jackson-databind dependency to 2.17.1 (#1486)"

This reverts commit 4a4f2ed767da749f48df26bd1aea73ee262a02ed.

dulceanu committed:
OAK-10833 - Consistency check reports success for repository with SNFE in checkpoints (#1487)

reschke committed:
Revert "Revert "OAK-10830: Upgrade jackson-databind dependency to 2.17.1 (#1486)""

This reverts commit a9d0c1c81295dfde2e3b1f1bb97af589e5238816.

mbaedke committed:
OAK-10621: o.a.j.o.namepath.JcrPathParser does not accept indexed expanded names (#1488)

* OAK-10621: o.a.j.o.namepath.JcrPathParser does not accept indexed expanded names.

Fixed, added unit tests.

reschke committed:
OAK-10824: Release Oak 1.64.0

nfsantos committed:
OAK-10825 - Indexing job: log [TASK:<PHASE>:FAIL] log message for when an indexing phase fails. (#1481)

reschke committed:
OAK-10838: oak-lucene: maven install fails in oak-lucene due to baseline check (#1491)

mbaedke committed:
OAK-6761: Convert oak-blob-plugins to OSGi R7 annotations (#1415)

done

t-rana committed:
Issues/oak 10781 (#1485)

* OAK-10675: add service principal support in oak-blob-cloud-azure

* OAK-10675: use user delegation key signed sas for service principal and add license to new files

* OAK-10675: some self review changes

* OAK-10675: set null blob headers as empty string

* OAK-10675: update pom to resolve package dependency

* OAK-10675: add import packages to resolve bundle

* OAK-10770: embed runtime dependencies of azure identity in oak-segment-azure

* OAK-10770: remove unused imports, refactor dependencies

* OAK-10770: add imports, refactor dependencies

* OAK-10675: fix runtime dependency issues, code review

* OAK-10781: add access token refresh mechanism

* close executor

* reduce token refresh delauy to 1 minute

* retrigger build

rishabhdaim committed:
OAK-10739 : Provide support for FullGC in Mongo DocumentStore (#1454)

Co-authored-by: Stefan Egli <stefanegli@apache.org>
Co-authored-by: Rishabh Kumar <diam@adobe.com>
Co-authored-by: Jose Cordero <corderob@adobe.com>
Co-authored-by: Daniel Iancu <iancu@adobe.com>

Full details in branch: DetailedGC/OAK-10199

stefan-egli committed:
OAK-10843 : temporarily disable flaky tests

adelina-marian committed:
OAK-10837: Add documentation for UT/IT NodeStore fixtures

shodaaan committed:
- changed check for node copied or moved over an existing node to compare the ":source-path" property for before and after nodes - this will ensure that check only returns true when the actual change is saved for a node
- added check for restoring version in VersionEditor - code path should not run when node is copied or moved because it would automatically check in the current node (by calling versionManager.restore) when propertyChanged is called for the "jcr:baseVersion" property. This would lead to the destination node (target of move or copy) to be checked in, which would cause "Node is checked-in" exceptions to appear for any subsequent property.put() calls on the node. The check for the after property to start with "restore-" is also necessary in order for restore version to work properly for the moved or copied node, after the move or copy is completed.
- removed unit tests with MEMORY_NS and SEGMENT_TAR fixtures, the ones with DOCUMENT_NS are the most relevant ones

stefan-egli committed:
OAK-10844 : removing some test combinations and adding logging to help narrow down where time is spent

dulceanu committed:
OAK-10807 - Add oak-run diff to Azure repositories (#1496)

stefan-egli committed:
OAK-10843 : exclude a flaky test combination

stefan-egli committed:
OAK-10844 : marking a slow test as ignored for now

stefan-egli committed:
OAK-10846 : ignoring testPartialMergeRootCleanup as it seems to leak a thread

fabriziofortino committed:
OAK-10841: log error cause when ES cannot process a query (#1492)

reschke committed:
OAK-10847: query-spi: bump log level to ERROR for use of deprecated slf4j.event.Level method (#1500)

reschke committed:
OAK-10852: oak-store-document: disable unreliable tests in BranchCommitGCTest

stefan-egli commented:
@shodaaan `MoveVersionableNodeWithNodeRepositoryTest` is missing the license header

shodaaan committed:
- added license header to MoveVersionableNodeWithNodeRepositoryTest

stefan-egli committed:
OAK-10853 : temporarily disable flaky test

stefan-egli commented on oak-core/src/main/java/org/apache/jackrabbit/oak/plugins/version/VersionEditor.java:
```diff
@@ -152,20 +153,37 @@ public void propertyChanged(PropertyState before, PropertyState after)
-                baseVersion = baseVersion.substring(RESTORE_PREFIX.length());
-                node.setProperty(JCR_BASEVERSION, baseVersion, Type.REFERENCE);
+            // OAK-8848: skip restore if this is an overwrite of a node and a restore operation is not in progress
+            if (!nodeWasMovedOrCopied() || after.getValue(Type.REFERENCE).startsWith(RESTORE_PREFIX)) {
```
I'm a bit lost at this condition here and wonder if it needs to be at this complexity. I'm also wondering if all permutations of this are covered in unit tests. Let's say nodeWasMovedOrCopied is false, why is the second part of the condition needed then? Can you please elaborate a bit on this?

shodaaan replied:
The reason why the nodeWasMovedOrCopied check is necessary in the code path that performs the version restore (starting with if (propName.equals(JCR_BASEVERSION))
) is that without it a bug occurs in certain conditions when doing the move with overwrite that is required to work in the definition of done of OAK-8848.

I will describe the flow that leads to the bug.

Prerequisites:
Start with two nodes, and create different versions for both of them. I tested this with image files that I then modify at least once, therefore creating versions for them.  

Then run the curl command for overwriting one of the nodes with the other. An example of using the Sling curl command for moving a node named Testfile.png from the parent sourceDir, under the parent destinationDir, where it overwrites another node with the same name is: 
curl -u admin:admin -X MOVE http://localhost:4502/api/assets/sourceDir/Testfile.png -H "X-Destination: http://localhost:4502/api/assets/destinationDir/Testfile.png" -H "X-Overwrite: T"  

If both of the nodes have version histories, i.e. they both have the jcr:versionHistory property set, as well as the jcr:baseVersion property set, the following flow will happen in the JCR layer of Oak, as a result of the session.save() call:

Step 1: ModifiedNodeState.compareAgainstBaseState will be called for the node Testfile.png (the moved node that will overwrite the previous node), leading to calls to EditorDiff.propertyChanged for all properties of the node.
This in turn will lead to VersionEditor.propertyChanged to be called for all properties of the  Testfile.png node.

Step 2: Eventually VersionEditor.propertyChanged will be called for the jcr:versionHistory property of the Testfile.png node. 
In my integration tests, this was called before being called for the jcr:baseVersion property. This order is very important, as will be shown in the following steps.
Since nodeWasMovedOrCopied will return true for Testfile.png, the throwProtected method will not be called.
This is the fix for the ConstraintViolationException that is described in OAK-8848.

Step 3: Since the node has multiple versions, it has the jcr:baseVersion property. Therefore, VersionEditor.propertyChanged will be called for this property as well.
When this happens, without the nodeWasMovedOrChanged check, the restore code will run, leading to versionManager.restore to be called for the Testfile.png node.

This logic is wrong, since there is no restore of a version of the node that should take place here.
Further, the VersionableState.restore method will be called for the Testfile.png node, leading to the call checkNotNull(versionable).setProperty(JCR_ISCHECKEDOUT, false, Type.BOOLEAN), which will set the jcr:isCheckedOut property for the node to false.
This is wrong behavior, since the move of a node over another should not result in the node being automatically checked in.

The fix I propose in the PR is to also add the check for nodeWasMovedOrChanged for the code that performs the restore. Since the check returns true, the restore logic will not run when the overwrite is performed.


Regarding the after.getValue(Type.REFERENCE).startsWith(RESTORE_PREFIX)) condition:
Debugging the code when restoring previous versions of a node led me to understand that the jcr:baseVersion property is first modified to have the prefix “restore-” to the original UUID, and then this prefix is used in the restore code section in VersionEditor.propertyChanged to set the jcr:baseVersion property back to its’ original UUID, when the method runs for the jcr:baseVersion property.
This leads me to believe that the check for startsWith(RESTORE_PREFIX) should also be added for clarity.


Regarding the unit tests, I agree that the bug described above should have a unit test that reproduces it. I will create it and update the PR.

stefan-egli replied:
Thx for the very detailed description!

> Regarding the after.getValue(Type.REFERENCE).startsWith(RESTORE_PREFIX)) condition:

As the code is now `restore()` is only called if this condition is true. Before this change however, it was also called in other cases. Now if those other cases were wrong, then that's a bug. I would however prefer to look into that and have a test cases that proofs that is a bug. I'm suspecting that with this change we'll introduce a regression...

shodaaan replied:
You are right, thank you for the comment. I will remove the second check.

nfsantos committed:
OAK-10851 -  Log START/END/FAIL messages of indexing job phases inside the same package (#1501)

Joscorbe committed:
OAK-10854: Allow oak.findOne to find long path documents

stefan-egli committed:
OAK-10852 : refinement of 46476bcbfc to only ignore flaky mode

reschke committed:
OAK-10856: oak-run: TextPopulatorTest uses transitive dependency com.beust.jcommander.internal.Maps (#1505)

reschke committed:
OAK-10852: disable BranchCommitGCTest unmergedAddsThenMergedAddThenUnmergedRemovesChildren for FullGCMode.ORPHANS_EMPTYPROPS_UNMERGED_BC (#1506)

stefan-egli committed:
OAK-10852 : excluding another flaky test combination

shodaaan committed:
- extracted into submethods the code that processes specific properties from VersionEditor.propertyChanged and added descriptions to the extracted methods
- cleaned up unit tests for overwrite node in the same session.save() and added checks for existing version history and adding property to the moved node - this covers the use case that require the extra nodeWasMovedOrCopied check for the JCR_BASEVERSION property check and processing

stefan-egli commented:
Regarding the refactoring of `propertyChanged` : so far I am not seeing the advantage of the refactoring. That code isn't reused elsewhere, it is merely cutting a larger method into 4 parts that have historically been together. I think the same could be achieved with adding more // comments where useful, the additional methods for me at least aren't improving things.

shodaaan committed:
Revert "- extracted into submethods the code that processes specific properties from VersionEditor.propertyChanged and added descriptions to the extracted methods"

This reverts commit 7886171f57e31f1f74974be8a2805f3b7ae56e9c.

reschke committed:
OAK-10860: oak-lucene: update gson test dependency and stop using deprecated APIs (#1509)

reschke committed:
OAK-10861: avoid use of Tika WriteOutHandler convenience method removed in Tika 2.x (#1510)

shodaaan committed:
- reverted the extracted methods from VersionEditor to the previous version of the code and kept the code explanations as comments

reschke commented on oak-jcr/src/test/java/org/apache/jackrabbit/oak/jcr/MoveVersionableNodeWithNodeRepositoryTest.java:
```diff
@@ -0,0 +1,230 @@
+            while (versionIterator.hasNext()) {
+                sourceNodeVersions.add(versionIterator.nextVersion());
+            }
+        }catch (Exception ex) {
```
```suggestion
        } catch (Exception ex) {
```

shodaaan replied:
Done.

reschke commented on oak-core/src/main/java/org/apache/jackrabbit/oak/plugins/version/VersionEditor.java:
```diff
@@ -145,27 +146,73 @@ public void propertyChanged(PropertyState before, PropertyState after)
                 vMgr.checkin(node);
             }
+
+       /*Completes the restore of a version from version history.
```
```suggestion
       /* Completes the restore of a version from version history.
```

shodaaan replied:
Done.

reschke commented on oak-core/src/main/java/org/apache/jackrabbit/oak/plugins/version/VersionEditor.java:
```diff
@@ -145,27 +146,73 @@ public void propertyChanged(PropertyState before, PropertyState after)
+       (see OAK-8848 for context), the restore operation must NOT be performed when the JCR_BASEVERSION property change
+       is processed for the node.
+
+       TODO: check if this code ever runs when after.getValue(Type.REFERENCE).startsWith(RESTORE_PREFIX) is FALSE,
```
Is this still a TODO?

shodaaan replied:
Done.

shodaaan replied:
No, it is not, I will remove the comment. I will remove the second check in the version restore code block - the one for after.getValue(Type.REFERENCE).startsWith(RESTORE_PREFIX) - since it is not necessary in order for all of the restore logic and the fix to work fine.

reschke commented on oak-jcr/src/test/java/org/apache/jackrabbit/oak/jcr/MoveVersionableNodeWithNodeRepositoryTest.java:
```diff
@@ -0,0 +1,230 @@
+ */
+public class MoveVersionableNodeWithNodeRepositoryTest extends AbstractRepositoryTest {
+
+    private static final String mixVersionable = "mix:versionable";
```
we should have a constant for this elsewhere... org.apache.jackrabbit.JcrConstants.MIX_VERSIONABLE;

shodaaan replied:
I used JcrConstants.MIX_VERSIONABLE. Done.

reschke commented on oak-jcr/src/test/java/org/apache/jackrabbit/oak/jcr/security/authorization/VersionManagementTest.java:
```diff
@@ -45,7 +45,7 @@ public class VersionManagementTest extends AbstractEvaluationTest {
     @Override
     @Before
-    protected void setUp() throws Exception {
+    public void setUp() throws Exception {
```
why are changes needed in this class?

shodaaan replied:
I reverted all of the changes, since the new unit tests are now in a different class.

shodaaan committed:
- switched hardcoded string to constant in new unit test and reverted changes in the VersionManagementTest unit test class
- made requested cosmetic changes

dulceanu committed:
OAK-9455 - Improve oak-run check to allow fast remote consistency checks (#1508)

nfsantos committed:
OAK-10868 - Use a constant to represent empty StringValue, to avoid allocating new objects in this case. (#1512)

nfsantos committed:
OAK-10866 - Avoid creating an intermediate String when appending a string value to a JsopBuilder. (#1511)

nfsantos committed:
OAK-10872 - Replace use of StringBuilder by string concatenation in PathUtils.concat(String, String)

stefan-egli committed:
OAK-10870 : New oak-mongo.js methods for sizes of properties, counts … (#1514)

* OAK-10870 : New oak-mongo.js methods for sizes of properties, counts by clusterId and unmerged branch commits

* OAK-10870 : addressed review comments

reschke reviewed (commented):
Thanks for the changes.

ionutzpi committed:
OAK-10803 - Modified DocumentPropertyState.java to compress values ba… (#1473)

Co-authored-by: pirlogea <pirlogea@adobe.com>

nscendoni committed:
OAK-10451 UserPrincipalProvider may cause many conflicts when under load (#1128)

Co-authored-by: anchela <angela@apache.org>
Co-authored-by: Marcel Reutegger <marcel.reutegger@gmail.com>

shodaaan committed:
- removed unnecessary condition from code block that completes version restore in VersionEditor
- made cosmetic changes to newly added comments

reschke committed:
OAK-10873: store-spi: remove use of Guava CharMatcher in test case (#1516)

* OAK-10873: store-spi: remove use of Guava CharMatcher in test case

* OAK-10873: store-spi: remove use of Guava CharMatcher in test case - use less expensive detection

* OAK-10873: store-spi: remove use of Guava CharMatcher in test case - off by one

* Revert "OAK-10873: store-spi: remove use of Guava CharMatcher in test case - off by one"

This reverts commit d78fe0af400f23ac4f1885b5cca6635981ebb94b.

anchela committed:
OAK-10451 : UserPrincipalProvider may cause many conflicts when under load (add missing licence header)

nfsantos committed:
OAK-10879 - Indexing job: In the index report, list phase timings in chronological order and not alphabetic order (#1519)

reschke committed:
Revert "OAK-10803 - Modified DocumentPropertyState.java to compress values ba… (#1473)"

This reverts commit 163352368343164671231b1ee8f4db75f3925877.

stefan-egli commented:
two more comments that I have drafted in [this temporary branch](https://github.com/apache/jackrabbit-oak/compare/OAK-8848-review-illustration?expand=1) for illustration:
* we could move the `nodeWasMovedOrCopied()` check for the 2nd case into that if block, to make it more limited to just that case. Previously it was throwing the exception - now I think we want it to not throw it anymore. With the check being in the outer if there is possibility it might still throw if the next if case hits (perhaps unlikely, but by doing it the above drafted way we'd reduce that risk)
* some reformatting suggestion to use `//` instead of `/*` - I think that's a bit more following how inline comments are frequently done

wdyt?

shodaaan committed:
- updated if checks that prevent constraint violation exception from being thrown
- cosmetic updates for added comments

shodaaan commented:
> two more comments that I have drafted in [this temporary branch](https://github.com/apache/jackrabbit-oak/compare/OAK-8848-review-illustration?expand=1) for illustration:
> 
> * we could move the `nodeWasMovedOrCopied()` check for the 2nd case into that if block, to make it more limited to just that case. Previously it was throwing the exception - now I think we want it to not throw it anymore. With the check being in the outer if there is possibility it might still throw if the next if case hits (perhaps unlikely, but by doing it the above drafted way we'd reduce that risk)
> * some reformatting suggestion to use `//` instead of `/*` - I think that's a bit more following how inline comments are frequently done
> 
> wdyt?

Thank you for the suggestions. I have made the changes in the latest commit.

reschke committed:
OAK-10883: document mk: disable flaky test VersionGarbageCollectorIT.testBundledPropUnmergedBCGC for mode ORPHANS_EMPTYPROPS_UNMERGED_BC (#1525)

stefan-egli commented:
Thx for the update @shodaaan 

I'm reviewing the changes and actually would like to find a test that verifies the opposite of the new MoveVersionableNodeWithNodeRepositoryTest to better understand the context etc. Might take me some more time. If someone else wants to chime in and approve in the meantime, pls feel free.

stefan-egli commented on oak-core/src/main/java/org/apache/jackrabbit/oak/plugins/version/VersionEditor.java:
```diff
@@ -145,27 +146,71 @@ public void propertyChanged(PropertyState before, PropertyState after)
         }
     }
 
+    /**
+     * Returns true if and only if the given node was moved or copied from another location.
+     */
+    private boolean nodeWasMovedOrCopied() {
```
@shodaaan a question about this one : I'm not sure this is also applicable to copy - I can see the move case, as that is covered through `:source-path` - but in which case is `:source-path` also added for copy?

shodaaan replied:
I just tested the copy instead of move case and it can not be done in one session, because WorkspaceDelegate.copy throws a javax.jcr.ItemExistsException. 
Copy was not mentioned in the ticket requirement either, it was a wrong assumption on my part that copy will behave the same as move in this case.

I will rename the check method to nodeWasMoved.

shodaaan committed:
Merge branch 'refs/heads/trunk' into OAK-8848

shodaaan committed:
- renamed check method to nodeWasMoved for better clarity

shodaaan committed:
Revert "- renamed check method to nodeWasMoved for better clarity"

This reverts commit fa2d241fdb7feddb4b1a39d40ae419d35d3ae4e9.

shodaaan committed:
Revert "Merge branch 'refs/heads/trunk' into OAK-8848"

This reverts commit 330fab27206d7a419aca928b479abc0ac958eeb7, reversing
changes made to 0980ced251d98e91ef36dc29591f8fafe66fd32f.

shodaaan committed:
- renamed check method for moved node in VersionEditor to nodeWasMoved

stefan-egli commented on RELEASE-NOTES.txt:
```diff
@@ -1,4 +1,4 @@
-Release Notes -- Apache Jackrabbit Oak -- Version 1.64.0
+Release Notes -- Apache Jackrabbit Oak -- Version 1.62.0
```
this file shouldn't be part of the PR

shodaaan replied:
Fixed it, thank you for the observation.

shodaaan committed:
- fixed version for RELEASE-NOTES.txt

shodaaan committed:
- fixed version for RELEASE-NOTES.txt

shodaaan committed:
- renamed newly created test to MoveVersionableNodeRepositoryTest

stefan-egli commented on oak-jcr/src/test/java/org/apache/jackrabbit/oak/jcr/MoveVersionableNodeRepositoryTest.java:
```diff
@@ -0,0 +1,229 @@
+        // check setting property - via NodeImpl - on moved node
+        setNodePropertyAndCheckIn(sourceNode, "testProp", "testValue");
+
+        // check version history - TODO HORIA: should be in separate method - EXTRACT
```
I would avoid personal TODOs and just have it without username..

shodaaan replied:
Fixed it, thank you for the catch.

stefan-egli reviewed (approved):
+1, lgtm now - one very minor comment that should block though.
I'll merge tomorrow to leave room for a potential other review

shodaaan committed:
- removed TODOs in test

stefan-egli commented:
jenkins failure seems unrelated to me and I'll go ahead and merge

in both oak-jcr and oak-store-document cases it is a failed fork :
```
[ERROR] org.apache.maven.surefire.booter.SurefireBooterForkException: The forked VM terminated without properly saying goodbye. VM crash or System.exit called?

[ERROR] Command was /bin/sh -c cd /home/jenkins/workspace/Jackrabbit_oak-trunk-pr_PR-1474/oak-jcr && /usr/local/asfpackages/java/adoptium-jdk-11.0.16.1+1/bin/java -javaagent:/home/jenkins/maven-repositories/1/org/jacoco/org.jacoco.agent/0.8.12/org.jacoco.agent-0.8.12-runtime.jar=destfile=/home/jenkins/workspace/Jackrabbit_oak-trunk-pr_PR-1474/oak-jcr/target/coverage-reports/jacoco-ut.exec -Xmx512m -XX:+HeapDumpOnOutOfMemoryError -Dupdate.limit=100 -Djava.awt.headless=true -jar /home/jenkins/workspace/Jackrabbit_oak-trunk-pr_PR-1474/oak-jcr/target/surefire/surefirebooter13470384701666492214.jar /home/jenkins/workspace/Jackrabbit_oak-trunk-pr_PR-1474/oak-jcr/target/surefire 2024-06-18T14-46-01_816-jvmRun1 surefire319862193111735272tmp surefire_012291488413358130927tmp
[ERROR] Error occurred in starting fork, check output in log
[ERROR] Process Exit Code: 143
[ERROR] Crashed tests:
[ERROR] org.apache.jackrabbit.oak.jcr.query.QueryTest
[ERROR] 	at org.apache.maven.plugin.surefire.booterclient.ForkStarter.fork(ForkStarter.java:669)
```
